components/jenkins/storage.tf

```
@@ -0,0 +1,15 @@
+resource "azurerm_storage_account" "jenkinsstg" {
+  name                     = "mgmtstatestore"
```

I think this is going to need to be another component, as you need to deploy it into every subscription that Jenkins is reaching too

it should also be

```suggestion
  name                     = "sdsstatestore${var.env}"
```